Reorganise reindexing operation

martynjames opened the pull request:
Rearrange reindexing button to use url formatted from within template; this will hopefully address the apparent flakiness of relying upon ```window.course``` member within javascript.

@dsego - Can you review as the first review?

dsego commented:
Looks good. Keep in mind there is a [PR #6914](https://github.com/edx/edx-platform/pull/6914) to change the reindex url.

martynjames commented:
@andy-armstrong - any chance you can take a quick look at this? Thanks for your help yesterday with this - I updated the template to include the url in the href attribute, the js code accordingly, and updated the tests.

andy-armstrong commented:
:+1: 

It's funny that you thought of passing the URL as the href of the button. It makes so much sense now that I've seen it, but I never pass URLs to the client that way.
